### PR TITLE
Use slimmer protobuf structures for remaining k8s structures

### PR DESCRIPTION
**Pull request by aanm**
Follow up of https://github.com/cilium/cilium/pull/11326 for remaining structures

### CODE REVIEW

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
test-me-please

**Comment by aanm** [comment]
Unable to debug k8s flakes due https://github.com/cilium/cilium/issues/11386

File: daemon/cmd/policy_test.go [hunk 1]
```diff
@@ -46,9 +47,7 @@ import (
 	envoy_api_v2_core "github.com/cilium/proxy/go/envoy/api/v2/core"
 	envoy_api_v2_route "github.com/cilium/proxy/go/envoy/api/v2/route"
 	envoy_type_matcher "github.com/cilium/proxy/go/envoy/type/matcher"
-
```
Comment (tgraf) [inline daemon/cmd/policy_test.go:49]: Your IDE is applying too much automation ;-)

**Review by tgraf — APPROVED** [review]

**Pull request merged**